app/controllers/fhir/r4/api_controller.rb

```diff
@@ -0,0 +1,493 @@
+      capabilities: ['launch-standalone']
+    }
+  end
+
+  # Handle OPTIONS requests for CORS preflight
+  def options
+    render plain: ''
+  end
+
+  private
+
+  # Current user account as authenticated via doorkeeper
+  def current_resource_owner
+    User.find(doorkeeper_token.resource_owner_id) if doorkeeper_token
+  end
+
+  # Current user account can use api
+  def check_api_access
+    current_resource_owner&.can_use_api?
+  end
+
+  # Check accept header for correct mime type (or allow fhir _format)
+  def accept_header?
+    return request.headers['Accept']&.include?('application/fhir+json') if params.permit(:_format)[:_format].nil?
+
+    ['json', 'application/json', 'application/fhir+json'].include?(params.permit(:_format)[:_format]&.downcase)
+  end
+
+  # Check content type header for correct mime type
+  def content_type_header?
+    request.headers['Content-Type']&.include?('application/fhir+json')
+  end
+
+  # Generic 406 not acceptable
+  def status_not_acceptable
+    respond_to do |format|
+      format.any { head :not_acceptable }
+    end
+  end
+
+  # Generic 415 unsupported media type
+  def status_unsupported_media_type
+    respond_to do |format|
+      format.any { head :unsupported_media_type }
+    end
+  end
+
+  # Generic 400 bad request response
+  def status_bad_request
+    respond_to do |format|
+      format.any { head :bad_request }
+    end
+  end
+
+  # Generic 403 forbidden response
+  def status_forbidden
+    respond_to do |format|
+      format.any { head :forbidden }
+    end
+  end
+
+  # Generic 201 created response
+  def status_created(resource)
+    respond_to do |format|
+      format.json { render json: resource.to_json, status: :created, location: full_url_helper(resource), content_type: 'applicaton/fhir+json' }
+      format.fhir_json { render json: resource.to_json, status: :ok, location: full_url_helper(resource), content_type: 'applicaton/fhir+json' }
+    end
+  end
+
+  # Generic 200 okay response
+  def status_ok(resource)
+    respond_to do |format|
+      format.json { render json: resource.to_json, status: :ok, content_type: 'applicaton/fhir+json' }
+      format.fhir_json { render json: resource.to_json, status: :ok, content_type: 'applicaton/fhir+json' }
+    end
+  end
+
+  # Get a patient by id
+  def get_patient(id)
+    current_resource_owner.viewable_patients.find_by(id: id)
+  end
+
+  # Search for patients
+  def search_patients(options)
+    query = current_resource_owner.viewable_patients
+    options.each do |option, search|
+      case option
+      when 'family'
+        query = query.where('last_name like ?', "#{search}%") if search.present?
+      when 'given'
+        query = query.where('first_name like ?', "#{search}%") if search.present?
+      when 'telecom'
+        query = query.where('primary_telephone like ?', "#{search}%") if search.present?
+      when 'email'
+        query = query.where('email like ?', "#{search}%") if search.present?
```

Comment:
I recommend these `"#{search}%"` queries be changed to `"%#{search}%"` -- notice the extra `%` at the beginning.

This will make search matches more lenient and not require the search term match only the start of the string.

Reply:
Agreed, I think that is definitely an improvement. Changed